core/domain/summary_services.py

```diff
@@ -68,9 +68,25 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids):
     Please use this function when needing summary information to display on
     exploration summary tiles in the frontend.
     """
-    displayable_exp_summaries = []
     exploration_summaries = (
         exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+
+    return get_displayable_exp_summary_dicts(exploration_summaries)
+
+
+def get_displayable_exp_summary_dicts(exploration_summaries):
+    """Given a list of exploration summary domain objects, returns
+     a list of the human-readable exploration summary dicts that
+     correspond to explorations that are currently non-private
+     and not deleted.
+    """
+    displayable_exp_summaries = []
+    exploration_ids = []
+
+    for exploration_summary in exploration_summaries:
```

Comment:
OK. You could actually replace the whole 'for loop' with something like:

```
    exploration_ids = [(
        exploration_summary.id if exploration_summary is not None else None
    ) for exploration_summary in exploration_summaries]
```

It would be nice to write a test covering this case, if you can.